pkg/channel/consolidated/status/status.go

```diff
@@ -0,0 +1,514 @@
+
+	ok, err := prober.Do(
+		ctx,
+		transport,
+		probeURL.String(),
+		opts...)
+
+	// In case of cancellation, drop the work item
+	select {
+	case <-item.context.Done():
+		m.workQueue.Forget(obj)
+		return true
+	default:
+	}
+
+	if err != nil || !ok {
+		// In case of error, enqueue for retry
+		m.workQueue.AddRateLimited(obj)
+		item.logger.Errorf("Probing of %s failed, IP: %s:%s, ready: %t, error: %v (depth: %d)",
+			item.url, item.podIP, item.podPort, ok, err, m.workQueue.Len())
+	} else {
+		m.onProbingSuccess(item.targetStates, item.podState)
+	}
+	return true
+}
+
+func (m *Prober) onProbingSuccess(subscriptionState *targetState, podState *podState) {
+	// The last probe call for the Pod succeeded, the Pod is ready
+	if podState.pendingCount.Dec() == 0 {
+		// Unlock the goroutine blocked on <-podCtx.Done()
+		podState.cancel()
+		// This is the last pod being successfully probed, the subscription is ready
+		if m.checkReadiness(subscriptionState) {
+			subscriptionState.cancel()
+			m.readyCallback(subscriptionState.ch, subscriptionState.sub)
+		}
+	}
+}
+
+func (m *Prober) onProbingCancellation(subscriptionState *targetState, podState *podState) {
+	for {
+		pendingCount := podState.pendingCount.Load()
+		if pendingCount <= 0 {
+			// Probing succeeded, nothing to do
+			return
+		}
+
+		// Attempt to set pendingCount to 0.
+		if podState.pendingCount.CAS(pendingCount, 0) {
+			// This is the last pod being successfully probed, the subscription is ready
+			if subscriptionState.pendingCount.Dec() == 0 {
+				subscriptionState.cancel()
+				m.readyCallback(subscriptionState.ch, subscriptionState.sub)
+			}
+			return
+		}
+	}
+}
+
+func (m *Prober) probeVerifier(item *workItem) prober.Verifier {
+	return func(r *http.Response, b []byte) (bool, error) {
+		m.logger.Debugw("Verifying response", zap.Int("status code", r.StatusCode),
+			zap.ByteString("body", b))
+		switch r.StatusCode {
+		case http.StatusOK:
+			/**
+			{"my-kafka-channel":["90713ffd-f527-42bf-b158-57630b68ebe2","a2041ec2-3295-4cd8-ac31-e699ab08273e","d3d70a79-8528-4df6-a812-3b559380cf08","db536b74-45f8-41cd-ab3e-7e3f60ed9e35","eb3aeee9-7cb5-4cad-b4c4-424e436dac9f"]}
+			*/
+			var subscriptions = make(map[string][]string)
+			err := json.Unmarshal(b, &subscriptions)
+			if err != nil {
+				m.logger.Errorw("error unmarshaling", err)
+				return false, err
+			}
+			uid := string(item.targetStates.sub.UID)
+			key := fmt.Sprintf("%s/%s", item.targetStates.ch.Namespace, item.targetStates.ch.Name)
+			m.logger.Debugw("Received proper probing response from target",
+				zap.Any("found subscriptions", subscriptions),
+				zap.String("pod ip", item.podIP),
+				zap.String("want channel", key),
+				zap.String("want subscription", uid),
+			)
+			if subs, ok := subscriptions[key]; ok && sets.NewString(subs...).Has(uid) {
+				item.targetStates.readyCount.Inc()
+				return true, nil
+			} else {
+				//TODO return and error if the channel doesn't exist?
```

Comment:
As chatted, lets add a log line here (perhaps info level?) so that we know wthat we've checked dispatcher foo, and haven't found it on that particular pod.

Reply:
while I was adding the log line, I noticed I actually have a log line just above it (before the if) that's giving all the info needed. so now I'm a bit reluctant to add yet another log line with almost same details. wdyt?